src/sentry/sentry_apps/tasks/sentry_apps.py

```
@@ -498,6 +491,7 @@ def send_resource_change_webhook(
 def notify_sentry_app(event: GroupEvent, futures: Sequence[RuleFuture]):
     for f in futures:
         if not f.kwargs.get("sentry_app"):
+            logger.info("notify_sentry_app.future_missing_sentry_app", extra={"future": f})
```

should we raise an exception? calling `action_inst.after` (which calls `notify_sentry_app` for sentry app actions) is wrapped in `safe_execute` so i believe raising an error will log an exception https://github.com/getsentry/sentry/blob/be3950b36c216d57e588af4503f2eb4b1b613bdb/src/sentry/rules/processing/processor.py#L164-L168

i suppose this means the action is broken anyway, so hopefully we wouldn't migrate it cc @iamrajjoshi

---

if one future is missing a sentry app does that mean the rest are broken? If we raise we'll stop prcessing all the other futures. I can capture here though if that works ?

---

i would be curious if we are ever missing a `future` in the first place, i have a suspicion that check exists because of mypy